data/basics/userBalance.go

```diff
@@ -76,6 +76,21 @@ func (s Status) String() string {
 	return ""
 }
 
+// UnmarshalStatus decodes string status value back to Status constant
+func UnmarshalStatus(value string) (s Status) {
+	switch value {
+	case "Offline":
+		s = Offline
+	case "Online":
+		s = Online
+	case "Not Participating":
+		s = NotParticipating
+	default:
+		s = Offline
```

Comment:
Shouldn't this case produce an error?

Reply:
It mirrors `Status.String()` and the fact `Offline` = 0 is default value

Reply:
I see. I think it might make sense to return an error to prevent the case where someone accidentally calls this on `"online"` and then gets `Offline`.

Reply:
I agree with @derbear, added an error here.